src/hpa.c

```diff
@@ -273,6 +272,37 @@ hpa_shard_stats_merge(
 	malloc_mutex_unlock(tsdn, &shard->grow_mtx);
 }
 
+static bool
+hpa_is_hugify_eager(hpa_shard_t *shard) {
+	return shard->opts.hugify_style == hpa_hugify_style_eager_and_lazy
+	    || shard->opts.hugify_style == hpa_hugify_style_eager;
+}
+
+static bool
+hpa_is_hugify_lazy(hpa_shard_t *shard) {
+	/* When Hugify_sync==true we also set/unset HG bit manually */
+	return shard->opts.hugify_style == hpa_hugify_style_eager_and_lazy
+	    || shard->opts.hugify_style == hpa_hugify_style_lazy
+	    || shard->opts.hugify_sync;
+}
+
+static bool
+hpa_is_hugify_none(hpa_shard_t *shard) {
+	return shard->opts.hugify_style == hpa_hugify_style_none;
+}
+
+/*
+ * We do not have a cheap way to be sure that certain hpadata that we believe is
+ * huge is indeed backed by hugepage.  Experimentation has shown that when we
+ * are purging only HUGEPAGE ranges and hugifying eagerly(or thp enabled=always)
+ * we get huge pages more often.  This helps us have more realistic accounting.
+ */
```

Comment:
We do have a way to get this information (cheaply?) by reading `/proc/self/pagemap`, right?  In a background thread, that should not be too costly.  Maybe just the first sentence can go away?